projects/plugins/jetpack/extensions/blocks/payments-intro/variations.js

```diff
@@ -1,42 +1,37 @@
 /**
  * The different payment blocks that can be chosen.
- *
- * Important note: This block must be loaded after the different blocks that can be chosen. This can be done by ensuring
- * this is at the bottom of the block list at `projects/plugins/jetpack/extensions/index.json`.
  */
 
-/**
- * External dependencies
- */
-import { getBlockType } from '@wordpress/blocks';
-
 /**
  * Internal dependencies
  */
-import { settings as donationSettings } from '../donations';
-import { settings as recurringPaymentSettings } from '../recurring-payments';
+import { name as premiumContentName, settings as premiumContentSettings } from '../premium-content';
+import { name as donationsName, settings as donationSettings } from '../donations';
+import {
+	name as recurringPaymentsName,
+	settings as recurringPaymentSettings,
+} from '../recurring-payments';
 
 const variations = [
-	[ 'donations', donationSettings ],
-	[ 'recurring-payments', recurringPaymentSettings ],
+	[ donationsName, donationSettings ],
+	[ recurringPaymentsName, recurringPaymentSettings ],
+	[ premiumContentName, premiumContentSettings ],
 ];
 
-const variationDefinitions = variations
-	.map( blockNameSettings => {
-		const [ blockName, settings ] = blockNameSettings;
-		if ( ! getBlockType( 'jetpack/' + blockName ) ) {
-			return {};
-		}
-		return {
-			name: 'payments-intro/' + blockName,
-			title: settings.title,
-			description: settings.description,
-			icon: settings.icon.src,
-			// The inner block itself is already listed in the inserter in its own right, so just include in this blocks
-			// unified intro.
-			scope: [ 'block' ],
-		};
-	} )
-	.filter( blockDefinition => Object.entries( blockDefinition ).length > 0 );
+const variationDefinitions = variations.map( blockNameSettings => {
+	let blockName = blockNameSettings[ 0 ];
+	const settings = blockNameSettings[ 1 ];
+
+	if ( ! blockName.includes( '/' ) ) {
+		blockName = 'jetpack/' + blockName;
+	}
+
```

Comment:
We could substitute this code for:
````
const variationDefinitions = variations.map( ( [ blockName, settings ] ) => {
	if ( ! blockName.includes( '/' ) ) {
		blockName = 'jetpack/' + blockName;
	}
````

That way we could increase readability by not needing to access elements of the current iteration by its index.

Reply:
Not a big fan of directly poking the arguments, which could lead to unexpected behaviours (such as mutating the original object).
In this case we could shorten it even further with a ternary:
```js
return {
  name: blockName.includes( '/' ) ? blockName : 'jetpack/' + blockName,
};
```